src/js/components/Report/Report.jsx

```diff
@@ -114,15 +114,20 @@ function Report({
 
   useEffect(() => {
     if (!fetched) {
-      httpGet(
-        state.fetch,
-        new URL(endpoint, state.baseURL),
-        ({ data }) => {
-          setReportData(data)
-          setErrorMessage(null)
-          setFetched(true)
+      fetchPages(
+        endpoint,
+        state,
+        (data, isComplete) => {
+          if (isComplete) {
+            setFetched(true)
+            setReportData((prevState) => prevState.concat(data))
+          } else {
+            setReportData((prevState) => prevState.concat(data))
+          }
```

Comment:
Small nit, I'd pull `setReportData` up since it always updates the data

```suggestion
          setReportData((prevState) => prevState.concat(data))
          if (isComplete) {
            setFetched(true)
```